upstream/cloud_provider/manifests/v1.26.1/release.yaml

```
@@ -0,0 +1,171 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: system:cloud-controller-manager
+roleRef:
+  kind: ClusterRole
+  name: system:cloud-controller-manager
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- apiGroup: ""
+  kind: ServiceAccount
+  name: cloud-controller-manager
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: system:cloud-controller-manager
+rules: 
+  - apiGroups:
+    - ''
+    resources:
+    - events
+    verbs:
+    - create
+    - patch
+    - update
+  - apiGroups:
+    - ''
+    resources:
+    - nodes
+    verbs:
+    - '*'
+  - apiGroups:
+    - ''
+    resources:
+    - nodes/status
+    verbs:
+    - patch
+  - apiGroups:
+    - ''
+    resources:
+    - services
+    verbs:
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups:
+    - ''
+    resources:
+    - services/status
+    verbs:
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups:
+    - ''
+    resources:
+    - serviceaccounts
+    verbs:
+    - create
+  - apiGroups:
+    - ''
+    resources:
+    - persistentvolumes
+    verbs:
+    - get
+    - list
+    - update
+    - watch
+  - apiGroups:
+    - ''
+    resources:
+    - endpoints
+    verbs:
+    - create
+    - get
+    - list
+    - watch
+    - update
+  - apiGroups:
+    - coordination.k8s.io
+    resources:
+    - leases
+    verbs:
+    - create
+    - get
+    - list
+    - watch
+    - update
+  - apiGroups:
+    - ''
+    resources:
+    - serviceaccounts/token
+    verbs:
+    - create
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: aws-cloud-controller-manager
+  labels:
+    k8s-app:  aws-cloud-controller-manager
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      k8s-app:  aws-cloud-controller-manager
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      name: aws-cloud-controller-manager
+      labels:
+        k8s-app:  aws-cloud-controller-manager
+    spec:
+      tolerations: 
+        - effect: NoSchedule
+          key: node.cloudprovider.kubernetes.io/uninitialized
+          value: 'true'
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/master
+        - effect: NoSchedule
+          key: node-role.kubernetes.io/control-plane
+      nodeSelector: 
+        node-role.kubernetes.io/control-plane: ''
+      dnsPolicy: Default
+      priorityClassName: system-node-critical
+      serviceAccountName: cloud-controller-manager
+      hostNetwork: true
+      securityContext: {}
+      containers:
+        - name: aws-cloud-controller-manager
+          image: "registry.k8s.io/provider-aws/cloud-controller-manager:v1.26.1"
+          args:
+            - --v=2
+            - --cloud-provider=aws
+            - --cluster-cidr={cluster_cidr}
```

I'm skeptical that --cluster-cidr is needed here. As I understand it, this is only needed in cloud controller managers that have --configure-cloud-routes=true, in which case they will divide up the cluster CIDR into blocks that get routed to each k8s node. We shouldn't need that in Charmed Kubernetes since we're using CNI implementations that handle routing themselves.

Can you drop --cluster-cidr and configure it with --configure-cloud-routes=false instead? Does that work?

---

sure, if that's possible and functional, we could drop these changes too
* https://github.com/juju-solutions/interface-kube-control/pull/39

---

Ah wait, I forgot about --allocate-node-cidrs. I think that needs to be false too.

---

I can give this a test deployment later today and poke at it if you like, rather than demanding you figure everything out

---

Ok yeah, grepping through the [cloud-provider-aws repo](https://github.com/kubernetes/cloud-provider-aws), I don't see any references to cluster-cidr or ClusterCIDR. I see the options defined in the generic cloud provider code [here](https://github.com/kubernetes/cloud-provider/blob/9be8e26d3703a1461004f30951315283a88307e4/options/kubecloudshared.go#L60-L63) and they seem to match with my understanding.

So, I think this should work:

```suggestion
            - --allocate-node-cidrs=false
            - --configure-cloud-routes=false
```